src/jlgen.jl

```diff
@@ -79,7 +79,6 @@ if VERSION >= v"1.11.0-DEV.1552"
 @inline function methodinstance(@nospecialize(ft::Type), @nospecialize(tt::Type),
                                 world::Integer=tls_world_age())
     sig = signature_type_by_tt(ft, tt)
-    @assert Base.isdispatchtuple(sig)   # JuliaLang/julia#52233
```

Comment:
The underlying function this ccalls may assert of segfault without this. There is probably a significant argument to be made that you should be using the public function `which(sig)` here, which doesn't have that footgun, but it currently doesn't know that it could optimize the `isdispatchtuple` case by calling `jl_method_lookup_by_tt`.